workshop_code/workshop_3/Workshop3.ipynb

```diff
@@ -0,0 +1,565 @@
+    "df_energy.drop('rv1', axis=1, inplace=True)\n",
+    "df_energy.drop('rv2', axis=1, inplace=True)\n",
+    "\n",
+    "# Inspect modified dataset\n",
+    "### FILL IN ###"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:16:54.019440Z",
+     "start_time": "2018-10-20T04:16:53.968849Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Sum appliances and lights for total energy\n",
+    "\n",
+    "### FILL IN ###\n",
+    "df_energy.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:16:55.050538Z",
+     "start_time": "2018-10-20T04:16:54.981300Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Then drop appliances and lights, as they are now redundant features\n",
+    "\n",
+    "df_energy.drop('Appliances', axis=1, inplace=True)\n",
+    "df_energy.drop('lights', axis=1, inplace=True)\n",
+    "df_energy.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:16:55.897261Z",
+     "start_time": "2018-10-20T04:16:55.848494Z"
+    },
+    "code_folding": []
+   },
+   "outputs": [],
+   "source": [
+    "# Split the dataset into features and labels for training, testing\n",
+    "\n",
+    "### FILL IN ###\n",
+    "df_energy_X.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-19T02:34:44.406540Z",
+     "start_time": "2018-10-19T02:34:44.393274Z"
+    },
+    "code_folding": []
+   },
+   "outputs": [],
+   "source": [
+    "df_energy_y.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:19.394285Z",
+     "start_time": "2018-10-20T04:33:19.068177Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Method that takes in dataframe, and degree of polynomial\n",
+    "# Outputs polynomial form of dataframe\n",
+    "\n",
+    "def PF(X, degree):\n",
+    "    X = np.asarray(X)\n",
+    "    tmp_raise = X\n",
+    "    for i in range(2, degree+1):\n",
+    "        tmp_raise = np.append(tmp_raise, np.power(X, i), axis=1)\n",
+    "    X = tmp_raise\n",
+    "    finalarr = np.ones((X.shape[0], X.shape[1]+1))\n",
+    "    finalarr[:,:-1] = X\n",
+    "    return finalarr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:19.920832Z",
+     "start_time": "2018-10-20T04:33:19.917660Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Do 60-20-20 train_test_split for training set, validation set, and test set\n",
+    "\n",
+    "def split(X, y):\n",
+    "    X_train, X_test, y_train, y_test = train_test_split(\n",
```

Comment:
Maybe add a comment here saying that the second one would split the Test set into test and val?

Reply:
sure, there are many cases like that, but I will explain it when we call the function to split the dataset into train, val, test sets. I do agree this could be a little cleaner tho :^)